hack/jenkins/e2e.sh

```diff
@@ -509,7 +509,8 @@ case ${JOB_NAME} in
     : ${GKE_API_ENDPOINT:="https://test-container.sandbox.googleapis.com/"}
     : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-test"}
     : ${E2E_NETWORK:="e2e-gke-test"}
-    : ${JENKINS_PUBLISHED_VERSION:="release/latest"}
```

Comment:
This is actually pretty explicit, so that the minute the release hits GCS, Jenkins picks it up. It means that the actual default server version from the test job can be a little more lazily set and is more of a dev convenience for us than anything relevant for Jenkins.